overlord/snapstate/handlers.go

```diff
@@ -309,20 +324,121 @@ func (m *SnapManager) installOneBaseOrRequired(st *state.State, snapName string,
 	return ts, err
 }
 
-func (m *SnapManager) installPrereqs(t *state.Task, base string, prereq []string, userID int, tm timings.Measurer) error {
+// updates a prerequisite, if it's not providing a content interface that a plug expects it to
+func updatePrereqIfOutdated(t *state.Task, snapName string, contentAttrs []string, userID int, flags Flags) (*state.TaskSet, error) {
+	st := t.State()
+
+	if len(contentAttrs) == 0 {
+		return nil, nil
+	}
+
+	if ok, err := shouldSkipToAvoidConflict(t, snapName); err != nil {
+		return nil, err
+	} else if ok {
+		return nil, nil
+	}
+
+	// check if the default provider can provide the content it's meant to
+	ok, err := hasAllContentAttrs(st, snapName, contentAttrs)
+	if err != nil {
+		return nil, err
+	}
+
+	if !ok {
+		// default provider is missing some content tags (likely outdated) so update it
+		ts, err := Update(st, snapName, nil, userID, flags)
+		if err != nil {
```

Comment:
as we do for the Install path, this needs to check whether the error is a conflict and return a Retry in that case